docs/model/index.md

```diff
@@ -0,0 +1,36 @@
+# farmOS Data Model
+
+## Goals
+
+The goal of the farmOS data model is to represent and organize a wide variety
+of agricultural and ecological systems to allow for easy access and analysis.
+This ranges from small-scale garden records to industrial agriculture to
+national forestry management.
+
+farmOS can model data that was recorded via manual data entry, as well as data
+streams from sensors or other applications.
+
+## Data types
```

Comment:
Thinking about taxonomy term types has got me thinking about this heading again. I would suggest we say "Resources" instead. This is coupling us a little closer to JSON:API, but we're already borrowing their concept of `attributes` and `relationships`, and I think "resources" is a more broadly articulated term in web dev.

This would round out our terminology quite nicely, I think:

```
A log is a resource.
An asset is a resouce.
An activity is a resource type, specifically a log type.
The Drupal entity implements the farmOS resource.
The Drupal bundle implements the farmOS resource type.
```

I think that resolves any collisions we might otherwise have in naming.

Reply:
I thought about using the word "resources" originally too! I like it! Ultimately I decided to go with "Data types" instead, simply because I thought that would be a bit more approachable to a non-technical audience. I like "resource" personally, but if you aren't an API programmer it could be very confusing. Then layer on the fact that "resources" in farming means "stuff you use to farm" - I worry it would be confusing.

At the same time, I think "attributes" and "relationships" are easy to understand.

What about instead of "data types" we say "record types"?

Reply:
> What about instead of "data types" we say "record types"?

Following your pattern:

```
A log is a record.
An asset is a record.
An activity is a record type, specifically a log type.
The Drupal entity implements the farmOS record.
The Drupal bundle implements the farmOS record type.
```

Reply:
Whaddya think? https://github.com/farmOS/farmOS/pull/415/commits/d5d8f10defd04bfd5cd4fa5ffa1dc49261e047ff

Reply:
Sounds good to me!